### PR TITLE
auth: merge "main" code into new auth-main.(cc|hh)

### PR DESCRIPTION
**Pull request by zeha**
### Short description

Merge common_startup.cc and receiver.cc into auth-main.cc, and
rename common_startup.hh into auth-main.hh.
This is a very minimal merge with no cleanup. Its intention is to
make understanding the startup code path easier, by avoiding having
to look at two compilation units to determine the exact order.

The other reason to exclude any other work is to keep "just do it again" simple, in case significant changes to the involved files happen before this can be merged. Also maybe it's not a wanted cleanup in the first place?

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

**Comment by Habbie** [comment]
> Also maybe it's not a wanted cleanup in the first place?

I like it! Will try to review and merge soon.

**Comment by Habbie** [comment]
Brief review looks good. If you can resolve the conflict (which I trust is an almost mechanical action) I can check a bit closer and then merge.

**Comment by zeha** [comment]
> Brief review looks good. If you can resolve the conflict (which I trust is an almost mechanical action) I can check a bit closer and then merge.

Rebased!

Note for future self: on rebasing, drop format commit, the "meaty" commit should rebase just fine. Then rerun clang-format -i and commit that.

File: pdns/auth-main.cc [hunk 1]
```diff
@@ -0,0 +1,1485 @@
...
+#endif
+typedef Distributor<DNSPacket, DNSPacket, PacketHandler> DNSDistributor;
+
+ArgvMap theArg;
+StatBag S; //!< Statistics are gathered across PDNS via the StatBag class S
+AuthPacketCache PC; //!< This is the main PacketCache, shared across all threads
+AuthQueryCache QC;
+AuthZoneCache g_zoneCache;
+std::unique_ptr<DNSProxy> DP{nullptr};
+std::unique_ptr<DynListener> dl{nullptr};
```
[the remark this answers is not in the document]
Comment (zeha) [reply pdns/auth-main.cc:120]: can fix in a followup

**Comment by zeha** [comment]
All of the new codeql warnings are preexisting warnings, just now moved into a new place. Some of them surely can be fixed in a follow-up PR.

**Comment by zeha** [comment]
Rebased to fix Makefile.am conflict. Actual .cc/hh files appear unchanged.

**Review by Habbie — APPROVED** [review]

**Pull request merged**